### ISSUE TITLE
MangledAtomicTypeNamePrefix option - is this used by anybody?

### ISSUE THREAD

**Issue by aratajew** [KhronosGroup/SPIRV-LLVM-Translator#514]
I'd like to ask if anybody is using `MangledAtomicTypeNamePrefix `translator option?

It is handled in this part of code:
https://github.com/KhronosGroup/SPIRV-LLVM-Translator/blob/master/lib/SPIRV/SPIRVToOCL.cpp#L365

`void SPIRVToOCL::translateMangledAtomicTypeName()` function is logically meaningless and it has a lot of bugs, example:
```
auto Loc = MangledName.find(kOCLBuiltinName::AtomPrefix);
Loc = MangledName.find(kMangledName::AtomicPrefixInternal, Loc);
MangledName.replace(Loc, strlen(kMangledName::AtomicPrefixInternal),
                        MangledAtomicTypeNamePrefix);
```
Loc stored two times. No check if Loc isn't string::npos.

This function makes translator crash on such a simple OpenCL C kernel:
```
void __attribute__((overloadable)) atom_custom_function(){}
void kernel test()
{
	atom_custom_function();
}
```

My questions are:
1. Is anybody using it?
      a)  If not, I'll totaly remove this code.
      b)  If yes, please let me know, I'll just fix this code to avoid crashes.

Thanks,
Andrzej

**Comment by aratajew** [comment]
@yxsamliu I think, you an author of this code. Is it still in use?

**Comment by aratajew** [comment]
@svenvh , @AlexeySachkov , @AlexeySotkin guys, if any of you know some customer of this code, please, let me know.

**Comment by svenvh** [comment]
> @svenvh , @AlexeySachkov , @AlexeySotkin guys, if any of you know some customer of this code, please, let me know.

No objections from my side to remove the option.

**Comment by yxsamliu** [comment]
> @yxsamliu I think, you're an author of this code. Is it still in use?

I am no longer working on this project, so I will let others to decide. Thanks.

**Comment by AlexeySachkov** [comment]
Fixed by #515
